dotCMS/src/main/java/com/dotcms/rest/api/v1/index/ESIndexResource.java

```diff
@@ -67,392 +67,351 @@
-
-        	ResponseUtil.handleAsyncResponse(
-        	        () -> {
-
-                        try {
-
-                            if(this.indexAPI.uploadSnapshot(inputFile)){
-                                return new MessageEntity(LanguageUtil.get(user.getLocale(),
-                                        "snapshot.upload.success"));
-                            }
-
-                            throw new ServiceUnavailableException("snapshot.upload.failed");
-                        } catch (InterruptedException | LanguageException | ExecutionException | IOException e) {
-
-                            Logger.error(ESIndexResource.class, e.getMessage(), e);
-                            throw new DotRuntimeException("snapshot.upload.failed", e);
-                        }
-                    },
-                    (Throwable e) -> {
-
-                        Logger.error(this, "Exception trying to restore index snapshot:" + e.getMessage(), e);
-                        return ResponseUtil.mapExceptionResponse(e);
-                    },
-                    asyncResponse);
-
-        } catch (Exception e) {
-            Logger.error(this.getClass(),"Exception trying to restore index snapshot: " + e.getMessage(), e);
-            asyncResponse.resume(ResponseUtil.mapExceptionResponse(e));
-        }
+    @CloseDBIfOpened
+    @DELETE
+    @JSONP
+    @NoCache
+    @Path("/failed")
+    @Produces({MediaType.APPLICATION_JSON})
+    public Response deleteFailedRecords(@Context final HttpServletRequest request, @Context final HttpServletResponse response)
+                    throws DotDataException {
+        final InitDataObject init = auth(request, response);
+        APILocator.getReindexQueueAPI().deleteFailedRecords();
+        return Response.ok(new ResponseEntityView(true)).build();
     }
 
+    
 
-    /**
-     * @deprecated  As of 2016-10-12, replaced by {@link #snapshotIndex(HttpServletRequest,HttpServletResponse ,String)}
-     */
+
+    @Deprecated
     @PUT
     @Path("/create/{params:.*}")
     @Produces("text/plain")
     public Response createIndex(@Context HttpServletRequest httpServletRequest, @Context final HttpServletResponse httpServletResponse, @PathParam("params") String params) {
         try {
-            InitDataObject init=auth(params,httpServletRequest, httpServletResponse);
+            InitDataObject init=auth(httpServletRequest, httpServletResponse);
 
             int shards=Integer.parseInt(init.getParamsMap().get("shards"));
             boolean live = init.getParamsMap().containsKey("live") ? Boolean.parseBoolean(init.getParamsMap().get("live")) : false;
             String indexName = init.getParamsMap().get("index");
 
-            indexName = create(indexName, shards, live);
+            if(indexName == null)
+                indexName=ContentletIndexAPIImpl.timestampFormatter.format(new Date());
+            indexName = (live) ? "live_" + indexName : "working_" + indexName;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Prefer StringBuilder (non-synchronized) or StringBuffer (synchronized) over += for concatenating strings](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)